rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -105,97 +110,100 @@ def get_energy_required_to_heat_swh_use(
         if space_id in space_within_building_segment_ids
     ]
 
-    if len(spaces) == 0 and use_units not in REQUIRED_USE_UNIT:
+    if not spaces and use_units not in REQUIRED_USE_UNIT:
         spaces = find_all("$.zones[*].spaces[*]", building_segment)
 
-    energy_required_by_space = {}
-    volume_flow_rate = ZERO.FLOW
     hourly_values = (
         hourly_schedule.get("hourly_values")
         if hourly_schedule is not None
         else [1] * len(inlet_temperature_hourly_values)
     )
+
+    swh_use_value = swh_use.get("use", 0.0)
+    energy_required_by_space = {}
+    volume = ZERO.VOLUME
     for space in spaces:
+        space_id = space["id"]
         if use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_PERSON:
-            power = swh_use_value * space.get("number_of_occupants", 0)
+            power = swh_use_value * ureg("W") * space.get("number_of_occupants", 0)
             energy_required = (
                 power
                 * sum(hourly_values)
+                * ureg("hr")
                 * (1 - drain_heat_recovery_efficiency)
-                * ureg("Btu")
             )
-            energy_required_by_space[space["id"]] = energy_required
+            energy_required_by_space[space_id] = energy_required
 
         elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_AREA:
-            power = swh_use_value * space.get("floor_area", ZERO.AREA).magnitude
+            power = swh_use_value * ureg("W/m2") * space.get("floor_area", ZERO.AREA)
```

Comment:
- otherwise if use_units is POWER_PER_AREA: `elsif use_units == "POWER_PER_AREA":`
    - set power equal to the swh_use_value * the floor area in the space: `power = swh_use_value * space.floor_area`
    - in this case, swh_use_value is assumed to have units of btu/hr/ft2, so calculate energy_required by multiplying power (btu/hr) by the sum of the hourly_schedule: `energy_required = power * sum(hourly_schedule) * (1-drain_heat_recovery_efficiency)`

Reply:
i) You don't need ```* ureg("ft")``` because ```floor_area``` already has the area unit (ft2).
ii) Please use the power density unit - ```W/ft2```. I initially thought ```W/m2```, but when I checked the pint_utils.py file, power density unit is W/ft2.

Reply:
Yeah floor area is in m2, but it's unclear what unit is used for the ```shw_use``` when the ```SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_AREA``` is used.

@weilixu  Would it be safe to assume that the power per area unit is in ```W/ft2```?